src/main/java/com/thealgorithms/backtracking/combinationSum.java

```diff
@@ -0,0 +1,75 @@
+package com.thealgorithms.backtracking;
+
+import java.util.*;
+
+/**
+ * Java program for solving the Combination Sum problem.
+ * Author: Yash Kesharwani
+ * File: CombinationSum.java
+ * Comments: This program finds all combinations of candidates that sum to the target.
+ */
+public class combinationSum {
+    /**
+     * Recursively find combinations that sum to the target.
+     *
+     * @param res       The list to store valid combinations.
+     * @param candidates The array of candidate numbers.
+     * @param target    The target sum to achieve.
+     * @param sum1      The current sum during the recursion.
+     * @param start     The starting index for candidates.
+     * @param list1     The current combination being constructed.
+     */
+    public void check(List<List<Integer>> res, int[] candidates, int target, int sum1, int start, List<Integer> list1) {
+        if (sum1 == target) {
+            res.add(new ArrayList<>(list1));
+            return;
+        } else if (sum1 > target) {
+            return;
+        }
+        for (int i = start; i < candidates.length; i++) {
+            sum1 += candidates[i];
+            list1.add(candidates[i]);
+            check(res, candidates, target, sum1, i, list1);
+            sum1 -= candidates[i];
+            list1.remove(list1.size() - 1);
+        }
+    }
+
+    /**
+     * Find all combinations of candidates that sum to the target.
+     *
+     * @param candidates The array of candidate numbers.
+     * @param target    The target sum to achieve.
+     * @return A list of combinations that sum to the target.
+     */
+    public List<List<Integer>> combinationsum(int[] candidates, int target) {
+        List<List<Integer>> res = new ArrayList<>();
+        int start = 0, sum1 = 0;
+        List<Integer> list1 = new ArrayList<>();
+        check(res, candidates, target, 0, 0, list1);
+        return res;
+    }
+
+    public static void main(String[] args) {
```

Comment:
Please remove main and add some JUnit tests (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

Reply:
Okay